src/mlpack/methods/hdbscan/hdbscan_main.cpp

```diff
@@ -0,0 +1,84 @@
+/**
+ * @file hdbscan_main.cpp
+ * @author Sudhanshu Ranjan
+ *
+ * Implementation of program to run HDBSCAN.
+ */
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/util/cli.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+
+#include "hdbscan.hpp"
+
+using namespace mlpack;
+using namespace mlpack::hdbscan;
+using namespace std;
+
+PROGRAM_INFO("HDBSCAN algorithm", "This program helps in clustering "
+    "and is useful when the clusters have variable density "
+    "\n\n"
+    "The output is saved in a column matrix");
+
+PARAM_STRING_OUT("output_file",
+                 "Output file. Stores cluster label of each point.",
+                 "o");
+PARAM_STRING_IN("input_file",
+                "Input dataset to cluster",
+                "i",
+                "noFileSelected");
+PARAM_FLAG("single_cluster", "Allow single cluster.", "s");
```

Comment:
Does this mean allow all the points to lie in a single cluster, or allow clusters with a single point?  I think it is the former but we should be clear in the documentation.